ACE/bin/PerlACE/Process_Unix.pm

```diff
@@ -656,13 +660,16 @@ sub check_return_value ($)
         return ($rc >> 8);
     }
     elsif (($rc & 0xff) == 0) {
-        $rc >>= 8;
-        return $rc;
+        return ($rc >> 8);
     }
 
     # Ignore NSK 16-bit completion code
     $rc &= 0xff if $is_NSK;
 
+    if (defined($opt_signum)) {
+        ${$opt_signum} = $rc;
```

Comment:
Related to the other comment I think we should the separate signal and coredump info, so the caller doesn't have to do bitwise operations.